### DISCUSSION TITLE
Asserting/setting individual list element contents

### DISCUSSION THREAD

**Discussion by jpluscplusm** [cue-lang/cue#1706] [Q&A] [answered]
Are there any idiomatic and elegant ways to assert that (and set?) the Nth element of a list is a certain value?

I've found 2 mechanisms thus far:

1. Un-opinionated prefix, open suffix

```
Mylist: [ <some elements > ]
Mylist: [ _, _, _, "element I want to assert/set", ... ]
```

2. Stringified-integer-indexed struct

```
Mylist: [ <some elements > ]
Mylist_struct: {
  for i,e in Mylist {
    "\(i+1)": e 
  }
  "4": "element I want to assert"
}
```

The first method seems to scale poorly the further into the list I want to assert. The 2nd has minor jankiness with the stringified indexes, and that (I /think/?) it's "assert-only", and doesn't "push" elements back to the original list, if they were unset.

What's the current state of the art in this area?

**Comment by myitcv** [comment, marked as answer]
I would do something like this: 

```cue
import "list"

l: 10

// Establsh a list of length l
x: [for i in list.Range(0, l, 1) { _ }]

// Assert that the element at index 5 is 4
x: [for i in list.Range(0, l, 1) { if i == 5 { 4 } }]
```

[Playground link](https://cuelang.org/play/?id=JSkM1nI5lsa#cue@export@cue)

**Comment by mpvl** [comment]
With the query extension (#165), we intend to allow something like:

```
Mylist: 20: "element I want to assert/set"
```
or even
```
Mylist:  [>=10 & <=20]: "element I want to assert/set"
```
